src/ecoscope-workflows-core/ecoscope_workflows_core/indexes.py

```diff
@@ -67,6 +75,13 @@
 ]
 
 
+TemporalIndexSelector = tuple([mapping["value"] for mapping in strftime_directives])
+TemporalIndexJsonSchemaExtra = {
+    "enum": [mapping["value"] for mapping in strftime_directives],
+    "enumNames": [mapping["display"] for mapping in strftime_directives],
+}
```

Comment:
enum/enunNames is apparently deprecated in RJSF, see https://github.com/wildlife-dynamics/ecoscope-workflows/issues/151

Which, if memory serves, is why we went with the [oneOf implementation like this](https://github.com/wildlife-dynamics/ecoscope-workflows/blob/2989e5b247ad1a6ca58d74a6477d8a19026c163a/src/ecoscope-workflows-core/ecoscope_workflows_core/jsonschema.py#L46-L55)

For consistency, I would suggest we leverage that existing definition. Combining this with the idea above of iterating on keys/values, I would suggest we do the following (also, with a snake case variable name since this is an instance of oneOf):

```suggestion
from ecoscope_workflows_core.jsonschema import oneOf

temporal_index_json_schema_extra = oneOf(
    const=[k for k in strftime_directives.keys()],
    title=[v for v in strftime_directives.values()],
).model_dump()
```

Reply:
I'm not certain if the above code runs, but I imagine if it doesn't, it's quite close to being able to.

Reply:
We may need to use the key `"oneOf"` as we do for the dashboard filters here

https://github.com/wildlife-dynamics/ecoscope-workflows/blob/2989e5b247ad1a6ca58d74a6477d8a19026c163a/src/ecoscope-workflows-core/ecoscope_workflows_core/jsonschema.py#L69

so maybe something like 

```python
temporal_index_json_schema_extra = {
    "oneOf": oneOf(
        const=[k for k in strftime_directives.keys()],
        title=[v for v in strftime_directives.values()],
    ).model_dump()
}
```